app/lib/project/MetapolatorProject.js

```diff
@@ -655,50 +645,56 @@ define([
     };
 
-    function exportInstance(io, project, masterName, dirName, precision) {
-        // returns a models/Controller
-        var model = project.open(masterName)
-          , master = model.query('master#' + masterName)
-          , glyphSet
-          , exportController
-          ;
-
-        // create a bare ufoV2 directory
-        io.mkDir(false, dirName);
-
-        // create dirName/metainfo.plist
-        io.writeFile(false, dirName+'/metainfo.plist'
-                                , plistLib.createPlistString(metainfoV2));
-
-        // fontforge requires a fontinfo.plist that defines unitsPerEm
-        io.writeFile(false, dirName+'/fontinfo.plist'
-                                , plistLib.createPlistString(minimalFontinfo));
+    _p.exportUFOInstance_chunk = function(masterName, dirName, precision) {
+        if (this.chunk_io[masterName] == undefined){
+            this.chunk_io[masterName] = new InMemory();
+        }
 
-        io.mkDir(false, dirName+'/glyphs');
-        io.writeFile(false, dirName+'/glyphs/contents.plist', plistLib.createPlistString({}));
+        if (this.chunk_exportController[masterName] == undefined){
+            this.chunk_exportController[masterName] = new UFOExportController(this.chunk_io[masterName], this, masterName, dirName, precision);
+        }
 
-        glyphSet = GlyphSet.factory(false, io, dirName+'/glyphs', undefined, 2);
+        var it = this.chunk_exportController[masterName].run_export_iteration();
+        if (it.done){
+            delete this.chunk_exportController[masterName];
+        }
+        return it;
+    };
 
-        exportController = new UFOExportController(master, model, glyphSet, precision);
-        exportController.do_export();
+    function exportUFOInstance(io, project, masterName, dirName, precision) {
+        var exportController = new UFOExportController(io, project, masterName, dirName, precision)
+          , done = false
+          ;
+        while (!done) {
+            done = exportController.run_export_iteration().done;
+        }
     }
 
     _p.exportInstance = function(masterName, targetFileName, precision){
         if (targetFileName.slice(-8) === '.ufo.zip'){
-            var zipped = this.getZippedInstance(masterName, targetFileName, precision, 'nodebuffer');
+            var zipped = this.getZippedInstance(masterName, targetFileName.split(".zip")[0], precision, 'nodebuffer');
             this._io.writeFile(false, targetFileName, zipped);
         } else if (targetFileName.slice(-4) === '.otf'){
             var otf = this.getOTFInstance(masterName, this);
             this._io.writeFile(false, targetFileName, otf);
         } else {
-            exportInstance(this._io, this, masterName, targetFileName, precision);
+            exportUFOInstance(this._io, this, masterName, targetFileName, precision);
         }
     };
 
     _p.getZippedInstance = function(masterName, targetDirName, precision, dataType) {
```

Comment:
Keep the old implementation of this. The side-effect of `this.chunk_io[masterName]` is very confusing. Since the ui already handles zipping we will do it there. That's the price to pay for having control over the export to update the progressbar.